Optimize channel_shuffle_op on GPU

xiaomengy opened the pull request:
Summary: Optimize channel_shuffle_op on GPU

Differential Revision: D10639281

BIT-silence commented:
Benchmark on GPU
N = 256, C = 256, H = 128, W = 128, group = 8,
NCHW: 84.82ms -> 38.84ms
NHWC: 81.83ms -> 39.01ms

bddppq commented:
triggered c2 rocm tests run: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/3081/

BIT-silence commented:
@pytorchbot retest this please

xiaomengy committed:
Optimize channel_shuffle_op on GPU (#13066)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13066

Optimize channel_shuffle_op on GPU

Differential Revision: D10639281

fbshipit-source-id: 341e4ddec8e7bfecdbaa7fd0cec72d7c1e20f1e6
